src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.OpenFileTracker.cs

```diff
@@ -185,11 +185,17 @@ private void TryOpeningDocumentsForNewCookie(uint cookie)
                                     {
                                         w.OnDocumentOpened(documentId, textContainer, isCurrentContext);
                                     }
-                                    else
+                                    else if (w.CurrentSolution.ContainsAdditionalDocument(documentId))
                                     {
-                                        Debug.Assert(w.CurrentSolution.ContainsAdditionalDocument(documentId));
                                         w.OnAdditionalDocumentOpened(documentId, textContainer, isCurrentContext);
                                     }
+                                    else
+                                    {
+                                        // TODO: implement the ability for analyze config documents to be opened against
+                                        // a live text editor. This is tracked by
+                                        // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/750120
+                                        Debug.Assert(w.CurrentSolution.ContainsAnalyzerConfigDocument(documentId));
```

Comment:
A good suggestion. My concern would be that the OnAnalyzerDocumentOpened method *is* a public API, and so I didn't want to even have a stub until it's actually got something. I'm not sure exactly when this will get implemented, but having a no-op method might confuse other hosts (like VS for Mac) and just result in a "we called this and it didn't work" bug. And then implementing it might make them crash. :smile:

Reply:
Couldn't we just make it not public until it's ready? Is it part of the API surface already?